aiohttp/connector.py

```diff
@@ -820,16 +826,19 @@ def __init__(
 
         self._use_dns_cache = use_dns_cache
         self._cached_hosts = _DNSCacheTable(ttl=ttl_dns_cache)
-        self._throttle_dns_events: Dict[Tuple[str, int], EventResultOrError] = {}
+        self._throttle_dns_futures: Dict[Tuple[str, int], Set[asyncio.Future[None]]] = (
+            {}
+        )
         self._family = family
         self._local_addr_infos = aiohappyeyeballs.addr_to_addr_infos(local_addr)
         self._happy_eyeballs_delay = happy_eyeballs_delay
         self._interleave = interleave
         self._resolve_host_tasks: Set["asyncio.Task[List[ResolveResult]]"] = set()
 
     def _close_immediately(self) -> List[Awaitable[object]]:
-        for ev in self._throttle_dns_events.values():
-            ev.cancel()
+        for future_set in self._throttle_dns_futures.values():
```

Comment:
Is it worth using itertools.chain() to flatten this?

Reply:
sounds good to me

f2f126a4a